### ISSUE TITLE
Cast to "type" type

### ISSUE THREAD

**Issue by philrz** [brimdata/zed#4714]
Repro is with Zed commit e6be73e.

## tl;dr

I found myself wanting to perform this cast, but it's not implemented yet.

```
$ zq -version
Version: v1.8.1-64-ge6be73ee

$ echo '"<int64>"' | zq -z 'yield type(this)' -
error("cast to type not implemented")
```

## Details

The high level use case is that I want to cast values to richer Zed types so I can use them in certain functions. I'll explain by first using a non-type example.

```
$ echo '{addr:"1.2.3.4", net:1.0.0.0/8}' | zq -z 'yield cidr_match(net, ip(addr))' -
true
```

i..e, `addr` started as a `string`, so I turned it into an `ip` so I could use it as an arg to `cidr_match()`.

That worked, so let's start from this:

```
$ echo '{value:1, dest_type:<int8>}' | zq -z 'yield cast(value, dest_type)' -
1(int8)
```

That works since dest_type was a `type` type out of the gate. But what if I do this?

```
$ echo '{value:1, dest_type:"<int8>"}' | zq -z 'yield cast(value, type(dest_type))' -
error({message:"shaper type argument is not a type",on:error("cast to type not implemented")})
```

@mattnibs did point out this works, though.

```
$ echo '{value:1, dest_type:"<int8>"}' | zq -z 'yield cast(value, parse_zson(dest_type))' -
1(int8)
```

As a user, I'd not thought to try that since the other casting functions work the way they do.

**Comment by philrz** [comment]
Verified in Zed commit 8a48163.

The cast attempted in the repro steps now succeeds.

```
$ zq -version
Version: v1.12.0-28-g8a48163f

$ echo '"<int64>"' | zq -z 'yield type(this)' -
<int64>
```

Thanks @mattnibs!
